backends/aoti/runtime/AotiBackend.cpp

```diff
@@ -0,0 +1,532 @@
+    AOTITensorHandle* input_handles, // array of input AOTITensorHandle; handles
+                                     // are stolen; the array itself is borrowed
+    size_t num_inputs,
+    AOTITensorHandle*
+        output_handles, // array for writing output AOTITensorHandle; handles
+                        // will be stolen by the caller; the array itself is
+                        // borrowed
+    size_t num_outputs,
+    AOTInductorStreamHandle stream_handle,
+    AOTIProxyExecutorHandle proxy_executor_handle);
+
+  AOTInductorModelContainerCreateWithDeviceFunc AOTInductorModelContainerCreateWithDevice = nullptr;
+  AOTInductorModelContainerDeleteFunc AOTInductorModelContainerDelete = nullptr;
+  AOTInductorModelContainerGetNumInputsFunc AOTInductorModelContainerGetNumInputs = nullptr;
+  AOTInductorModelContainerGetNumOutputsFunc AOTInductorModelContainerGetNumOutputs = nullptr;
+  AOTInductorModelContainerRunFunc AOTInductorModelContainerRun = nullptr;
+  std::unordered_map<Tensor*, std::vector<int64_t>> tensor_to_sizes;
+  std::unordered_map<Tensor*, std::vector<int64_t>> tensor_to_strides;
+  std::unordered_set<std::shared_ptr<Tensor>> tensors;
+
+  int32_t aoti_torch_grad_mode_is_enabled() {
+    // No autograd ever
+    return false;
+  }
+  void aoti_torch_grad_mode_set_enabled(bool enabled) {
+    if (enabled) {
+        std::runtime_error("Cannot enable autograd");
+    }
+  }
+  AOTITorchError aoti_torch_get_data_ptr(
+    AOTITensorHandle tensor,
+    void** ret_data_ptr) {
+    *ret_data_ptr = tensor->mutable_data_ptr();
+   return Error::Ok;
+  }
+  AOTITorchError aoti_torch_get_storage_offset(
+    AOTITensorHandle tensor,
+    int64_t* ret_storage_offset) {
+    // Storage offset is always 0 in ET
+    *ret_storage_offset = 0;
+    return Error::Ok;
+  }
+  AOTITorchError aoti_torch_get_strides(
+    AOTITensorHandle tensor,
+    int64_t** ret_strides) {
+    if (tensor_to_strides.find(tensor) == tensor_to_strides.end()) {
+      std::vector<int64_t> strides(tensor->dim());
+      for (int i = 0; i < tensor->dim(); i++) {
+        strides[i] = tensor->strides()[i];
+      }
+      tensor_to_strides[tensor] = strides;
+    }
+    *ret_strides = tensor_to_strides[tensor].data();
+    std::cout << "getting strides from tensor " << tensor << " with dim " << tensor->dim() << std::endl;
+    for (int i = 0; i < tensor->dim(); i++) {
+      std::cout << "strides " << i << " = " << *ret_strides[i] << std::endl;
+    }
+    return Error::Ok;
+  }
+
+  AOTITorchError aoti_torch_get_dtype(
+    AOTITensorHandle tensor,
+    int32_t* ret_dtype) {
+    *ret_dtype = static_cast<int32_t>(tensor->scalar_type());
+    return Error::Ok;
+  }
+
+  AOTITorchError aoti_torch_get_sizes(
+    AOTITensorHandle tensor,
+    int64_t** ret_sizes) {
+    if (tensor_to_sizes.find(tensor) == tensor_to_sizes.end()) {
+      std::vector<int64_t> sizes(tensor->dim());
+      for (int i = 0; i < tensor->dim(); i++) {
+        sizes[i] = tensor->size(i);
```

Comment:
why do we use the slow wrapping size() accessor here but the fast strides() accessor above?